src/compiler/moduleNameResolver.ts

```diff
@@ -961,7 +966,10 @@ namespace ts {
                 trace(state.host, Diagnostics.paths_option_is_specified_looking_for_a_pattern_to_match_module_name_0, moduleName);
             }
             const baseDirectory = getPathsBasePath(state.compilerOptions, state.host)!; // Always defined when 'paths' is defined
-            return tryLoadModuleUsingPaths(extensions, moduleName, baseDirectory, paths, loader, /*onlyRecordFailures*/ false, state);
+            if (!state.compilerPathPatterns) {

```

Comment:
Locally, I changed this to
```ts
            const anyOpts = state.compilerOptions as any;
            if (!anyOpts.compilerPathPatterns) {
                anyOpts.compilerPathPatterns = tryParsePatterns(paths);
            }
            return tryLoadModuleUsingPaths(extensions, moduleName, baseDirectory, paths, anyOpts.compilerPathPatterns, loader, /*onlyRecordFailures*/ false, state);
```

It was just a lazy way to store one per CompilerOptions.  Project load time dropped from 62s to 38s (greater than the total cost of parsing because it also reduced GC time by 5s).

Reply:
You can add this field on `options.configFile?.configFileSpecs?.pathPatterns`

Reply:
@sheetalkamat Does that mean I should move the validation from `verifyCompilerOptions` in `program.ts` to `getConfigFileSpecs` in `commandLineParser.ts`?

Reply:
Dont move `verifyCompilerOptions`. That needs to be in program..

Reply:
Sorry, I think my question was unclear.  It looks like "includes" and "excludes" are validated in `getConfigFileSpecs`, whereas "paths" is validated in `verifyCompilerOptions`.  I'm not sure why the validation is happening in two places, but the most obvious distinguishing characteristic is that the ones in `getConfigFileSpecs` end up in the config file specs.  If the path patterns will now be in the config file specs, does that mean I should move their validation out of `verifyCompilerOptions` and into `getConfigFileSpecs`?

Reply:
No it does not move into `getConfigFileSpecs` since you want to verify paths options even if program was created with compiler options that didnt come from parsing the config file.. But was set programatically by api user

Reply:
Isn't that true for "includes" and "excludes" as well?

Where do I initialize the config file specs then?  Where I'm doing it in this PR (i.e. on first use)?

Reply:
yes on first use.. includes and excludes can only be specified in ConfigFileSpec.. They are not compilerOption. But paths is .

Reply:
I believe I've implemented your suggestion.